admin/client/components/ItemsTable.js

```diff
@@ -1,20 +1,149 @@
+		if(props.rowAlert.success || props.rowAlert.fail) {
+			CurrentListStore.rowAlert('reset');
+		}
+		
+		const dragged = monitor.getItem().index;
+		const over = props.index;
+
+		// self
+		if (dragged === over) {
+		  return;
+		}
+
+		props.moveItem(dragged, over);
+		monitor.getItem().index = over;
+	}
+};
+
+/**
+ * Specifies the props to inject into your component.
+ */
+function collect(connect, monitor) {
+  return {
+    connectDragSource: connect.dragSource(),
+    isDragging: monitor.isDragging(),
+    connectDragPreview: connect.dragPreview(),
+  };
+}
+
+function connect(connect) {
+	 return {
+		 connectDropTarget: connect.dropTarget()
+	};
+};
+
+const ItemsRow = React.createClass({
 	propTypes: {
 		columns: React.PropTypes.array,
-		items: React.PropTypes.array,
+		items: React.PropTypes.object,
 		list: React.PropTypes.object,
+		index: React.PropTypes.number.isRequired,
+		id: React.PropTypes.any.isRequired,
+		// Injected by React DnD:
+		isDragging: React.PropTypes.bool.isRequired,
+		connectDragSource: React.PropTypes.func.isRequired,
+		connectDropTarget: React.PropTypes.func.isRequired,
+		connectDragPreview: React.PropTypes.func.isRequired
 	},
 	deleteItem (item, e) {
 		if (!e.altKey && !confirm('Are you sure you want to delete ' + item.name + '?')) return;
 		CurrentListStore.deleteItem(item);
 	},
+	renderRows (item) {
+		let itemId = item.id;
+		let rowClassname = classnames({
+			'ItemList__row--sortable': this.props.isDragging,
+			'ItemList__row--selected': this.props.checkedItems[itemId],
+			'ItemList__row--manage': this.props.manageMode,
+			'ItemList__row--success': this.props.rowAlert.success === itemId,
+			'ItemList__row--failure': this.props.rowAlert.fail === itemId,
+		});
+		var cells = this.props.columns.map((col, i) => {
+			var ColumnType = Columns[col.type] || Columns.__unrecognised__;
+			var linkTo = !i ? `/keystone/${this.props.list.path}/${itemId}` : undefined;
+			return <ColumnType key={col.path} list={this.props.list} col={col} data={item} linkTo={linkTo} />;
+		});
+		// add sortable icon when applicable
+		if (this.props.list.sortable) {
+			cells.unshift(<ListControl key="_sort" type="sortable" dragSource={this.props.connectDragSource} />);
+		}
+		// add delete/check icon when applicable
+		if (!this.props.list.nodelete) {
+			cells.unshift(this.props.manageMode ? (
+				<ListControl key="_check" type="check" active={this.props.checkedItems[itemId]} />
+			) : (
+				<ListControl key="_delete" onClick={(e) => this.deleteItem(item, e)} type="delete" />
+			));
+		}
+		
+		var style = this.props.isDragging ? {
+			borderLeft: '2px solid #1385e5',
+			color: 'black',
+			backgroundColor: 'white',
+			cursor: 'move',
+			opacity: 0
+		} : {};		
+		return (
+			this.props.connectDragPreview(this.props.connectDropTarget(<tr style={{ ...style }} key={'i' + item.id} onClick={this.props.manageMode ? (e) => this.props.checkTableItem(item, e) : null} className={rowClassname}>{cells}</tr>))
```

Comment:
we probably only need to do this when `this.props.isDragging`? there may be some other calls in this file that can be similarly optimised. Not sure that I'm correct though, don't have a lot of experience with react-dnd (yet)